actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -79,11 +79,13 @@ def http_basic_authenticate_with(name:, password:, realm: nil, **options)
         end
 
         def http_basic_authenticate_or_request_with(name:, password:, realm: nil, message: nil)
+          name = name.to_s
+          password = password.to_s
           authenticate_or_request_with_http_basic(realm, message) do |given_name, given_password|
             # This comparison uses & so that it doesn't short circuit and
             # uses `secure_compare` so that length information isn't leaked.
-            ActiveSupport::SecurityUtils.secure_compare(given_name, name) &
-              ActiveSupport::SecurityUtils.secure_compare(given_password, password)
+            ActiveSupport::SecurityUtils.secure_compare(given_name.to_s, name) &
```

Comment:
Is there a reason `name.to_s` isn't called here instead?
```suggestion
            ActiveSupport::SecurityUtils.secure_compare(given_name.to_s, name.to_s) &
```

Reply:
It's a bit handwavy, but the idea was to cast to string once and for all at boot, so that if somehow you passed something that acts as a string but isn't, we don't open a timing attack (because presumably `to_s` would take longer, the longer the password is).

But again it wasn't to prevent a specific scenario, just to be extra careful. Anyway I refactored the PR, so this `to_s` is no more.